### PR TITLE
Log error message in CSI VSC when in error state

### PR DESCRIPTION
**Pull request by anshulahuja98**
Fixes: https://github.com/vmware-tanzu/velero/issues/5566

### CODE REVIEW

File: internal/util/util.go [hunk 1]
```diff
@@ -175,6 +175,9 @@ func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1api.VolumeSnap
 		// bound to the existing snapshot.
 		if snapshotContent.Status == nil || snapshotContent.Status.SnapshotHandle == nil {
 			log.Infof("Waiting for volumesnapshotcontents %s to have snapshot handle. Retrying in %ds", snapshotContent.Name, interval/time.Second)
+			if snapshotContent.Status == nil && snapshotContent.Status.Error != nil {
```
Comment (blackpiglet) [inline internal/util/util.go:178]: Looks like the if condition check should be 
`if snapshotContent.Status != nil && snapshotContent.Status.Error != nil`

File: internal/util/util.go [hunk 2]
```diff
@@ -183,7 +186,11 @@ func GetVolumeSnapshotContentForVolumeSnapshot(volSnap *snapshotv1api.VolumeSnap
 
 	if err != nil {
 		if err == wait.ErrWaitTimeout {
-			log.Errorf("Timed out awaiting reconciliation of volumesnapshot %s/%s", volSnap.Namespace, volSnap.Name)
+			if snapshotContent.Status == nil && snapshotContent.Status.Error != nil {
```
Comment (blackpiglet) [inline internal/util/util.go:189]: Looks like the if condition check should be
`if snapshotContent.Status == nil && snapshotContent.Status.Error != nil`

Comment (anshulahuja98) [reply internal/util/util.go]: Resolved

Comment (anshulahuja98) [reply internal/util/util.go]: != as per other message resolved

**Review by blackpiglet — APPROVED** [review]

**Comment by blackpiglet** [comment]
Could you squash the 3 commits into one?

**Comment by anshulahuja98** [comment]
> Could you squash the 3 commits into one? The DCO check failure is due to no signoff in the commit message. You can resolve this by adding `--signoff` parameter in the `git commit` command.

- Squashed the commits
- Signed off the commit
- Added changelog

**Review by shubham-pampattiwar — APPROVED** [review]

**Review by blackpiglet — APPROVED** [review]

**Pull request merged**